src/main/java/com/emc/object/s3/request/PutObjectTaggingRequest.java

```diff
@@ -0,0 +1,70 @@
+package com.emc.object.s3.request;
+
+import com.emc.object.EntityRequest;
+import com.emc.object.Method;
+import com.emc.object.s3.bean.ObjectTagging;
+import com.emc.object.util.RestUtil;
+
+import java.util.Map;
+
+public class PutObjectTaggingRequest extends S3ObjectRequest implements EntityRequest {
+
+    private ObjectTagging tagging;
+    private String versionId;
+
+    public PutObjectTaggingRequest(String bucketName, String key) {
+        super(Method.PUT, bucketName, key, "tagging");
+    }
+
+    @Override
+    public Map<String, String> getQueryParams() {
+        Map<String, String> queryParams = super.getQueryParams();
+        if (versionId != null) queryParams.put("versionId", versionId);
+        return queryParams;
+    }
+
+    @Override
+    public Long getContentLength() {
+        return null; // assume buffering
+    }
+
+    @Override
+    public boolean isChunkable() {
+        return false;
+    }
+
+    @Override
+    public Object getEntity() {
+        return tagging;
+    }
+
+    @Override
+    public String getContentType() {
+        return RestUtil.TYPE_APPLICATION_XML;
+    }
+
+    public ObjectTagging getTagging() {
```

Comment:
please pay attention to existing getter/setter/builder method order and be consistent - other request classes have getter/setter pairs first, then with* methods after that

Reply:
Changed